cmd/internal/cli/pull.go

```diff
@@ -116,6 +122,10 @@ func pullRun(cmd *cobra.Command, args []string) {
 		name = PullImageName
 	}
 
+	if PullDir != "" {
+		name = filepath.Join(PullDir, name)
```

Comment:
Question: Is it intended that this might prepend this to absolute paths?

Example: `PULLDIR=/home/user singularity pull /tmp/foo.sif library://ubuntu` --> `/home/user/tmp/foo.sif`

Doesn't seem right.

Reply:
Yeah, that behavior is correct. What doesn't seem right about it? What do you think it _should_ do?

Reply:
Well imagine a setup where you have `export SINGULARITY_PULLDIR=$HOME/.pulls` in your bashrc. Now multiple cases:
- `singularity pull library://ubuntu` -> `~/.pulls/ubuntu.sif`
- `singularity pull library://ubuntu` ubuntu-new -> `~/.pulls/ubuntu-new`
- `singularity pull library:// ubuntu ~/i-want-it-here/ubuntu.sif` -> ~/.pulls/home/user/i-want-it-here/ubuntu.sif`

The last is especially surprising if the user wasn't aware of PULLDIR being set.

So what it *should* do: Complete a relative path with PULLDIR, leave absolute paths unchanged. Essentially mimic `cd $PULLDIR; singularity pull [...]; cd -`